libs/wingc/src/parser.rs

```diff
@@ -2562,6 +2552,32 @@ impl<'s> Parser<'s> {
 	}
 }
 
+/// Get actual child by field name when multiple children exist because of extra nodes.
+/// It should be safe to use this instead of `node.get_child_by_field_name`
+/// in cases where there's doubt.
+fn get_actual_child_by_field_name<'a>(node: Node<'a>, field_name: &str) -> Option<Node<'a>> {
```

Comment:
Is the description saying that multiple children can have the same field name? I'm not sure I understand how the behavior of this function is different from `calling node.child_field_by_name(name)` (maybe a unit test would make it clearer)

Reply:
I'll give a detailed explanation to make sure I understand correctly :) 
The tree-sitter API is indeed confusing. A node in tree-sitter can be a named node or unnamed node depending on how the grammar was defined. Also a node can have a **field** name (not to be confused with a named node).
* `node.get_children_by_field_name(field_name)` - returns all (can be multiple) children with the specified field name. e.g. in our grammar an interface definition can have multiple children with the "extends" field name. These children can be **unnamed** nodes too (usually not that useful). e.g. in our grammar "extends" nodes also include nodes for the commas separating the extend types: `interface I extends A,B {}` will have three sibling nodes with the field name "extends": `A`,`,`,`B`.
* `node.get_child_by_field_name(field_name)` - will return the first (or `None`) of the children returned by `get_children_by_field_name` above. This is very useful in places in the grammar where we can assume only a single child. Like in a class "extends" (vs an interface "extends").
* `get_actual_children_by_field_name(node, field_name)` - Will return all (can be multiple) children that are **named**, this filters out the the comma nodes in the example above.
* `get_actual_child_by_field_name(node, field_name)` - Will return the the fist (or `None`) of the children returned by `get_actual_children_by_field_name`, this is the most useful function since it filters both on a field name and named nodes and assumes a single result. In most places in the grammar this should work. e.g. a binary expression node `a + b` has a single **named** `operator` node with a field name called `op`.

Reply:
Added a unit test for `get_actual_children_by_field_name`